Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -569,6 +569,267 @@ class UpstreamIntegrationTests: TestBase {
         assertExpectedEvents(ignoreUnexpectedEvents: true)
     }
 
+    // MARK: datastream config overrides test
```

Comment:
Looks like you added a bunch of test cases for this feature, what do you think about moving them in a separate file ie ConfigOverride_UpstreamIntegrationTests?

Reply:
Yes, I was debating, should we have a new file, but it seems like a good idea since we have bunch of test cases.